lib/logstash/outputs/s3/file_repository.rb

```diff
@@ -13,6 +13,9 @@ class FileRepository
         # Ensure that all access or work done
         # on a factory is threadsafe
         class PrefixedValue
+
+          attr_reader :stale_time
+
           def initialize(file_factory, stale_time)
             @file_factory = file_factory
             @lock = Mutex.new
```

Comment:
We can make this a reentrant lock:

```suggestion
            @lock = Monitor.new
```